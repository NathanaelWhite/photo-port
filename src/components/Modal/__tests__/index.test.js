import React from 'react';
import { render, cleanup, fireEvent, getByTestId } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Modal from '..';

const mockToggleModal = jest.fn();
const currentPhoto = {
    name: 'Park Bench',
    category: 'landscape',
    description: 'a bunch of words the describe the photo you looking at fool',
    index: 1
};

afterEach(cleanup);

describe('Modal component', () => {
    
    it('renders', () => {
        render (<Modal 
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
            />)
    });

    it('matches snapshot', () => {
        const { asFragment } = render (<Modal 
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
            />)
        expect(asFragment()).toMatchSnapshot();
    });
})

describe('click event', () => {
    it('calls onClsoe handler', () => {
        // Arrange: render modal
        const { getByText } = render(<Modal
            onClose={mockToggleModal}
            currentPhoto={currentPhoto}
            />);
        // Act: Simulate click event
        fireEvent.click(getByText('Close this modal'));
        // Assert: expect matcher
        expect(mockToggleModal).toHaveBeenCalledTimes(1);
    });
})

