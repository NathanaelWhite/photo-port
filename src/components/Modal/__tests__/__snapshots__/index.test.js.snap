// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="modalBackdrop"
  >
    <div
      class="modalContainer"
    >
      <h3
        class="modalTitle"
      >
        Park Bench
      </h3>
      <img
        alt="current category"
      />
      <p>
        a bunch of words the describe the photo you looking at fool
      </p>
      <button
        type="button"
      >
        Close this modal
      </button>
    </div>
  </div>
</DocumentFragment>
`;
